amundsen_application/static/js/ducks/search/filters/reducer.ts

```diff
@@ -0,0 +1,119 @@
+import { FilterInput, ResourceType } from 'interfaces';
+
+export enum UpdateSearchFilter {
+  ADD_MULTI_SELECT = 'amundsen/search/filter/ADD_MULTI_SELECT',
+  REMOVE_MULTI_SELECT = 'amundsen/search/filter/REMOVE_MULTI_SELECT',
+  UPDATE_SINGLE =  'amundsen/search/filter/UPDATE_SINGLE',
+}
+export interface UpdateSearchFilterAction {
+  payload: FilterInput;
+  type: UpdateSearchFilter;
+};
+
+/* ACTIONS */
+export function updateSingleOption(input: FilterInput) {
+  const { category, value } = input;
+  return {
+    payload: {
+      category,
+      value
+    },
+    type: UpdateSearchFilter.UPDATE_SINGLE,
+  };
+};
+export function addMultiSelectOption(input: FilterInput) {
+  const { category, value } = input;
+  return {
+    payload: {
+      category,
+      value
+    },
+    type: UpdateSearchFilter.ADD_MULTI_SELECT,
+  };
+};
+export function removeMultiSelectOption(input: FilterInput) {
+  const { category, value } = input;
+  return {
+    payload: {
+      category,
+      value
+    },
+    type: UpdateSearchFilter.REMOVE_MULTI_SELECT,
+  };
+};
+
+/* REDUCER */
+export type FilterCategories = { [id:string]: string };
+export interface FilterReducerState {
+  [ResourceType.table]: TableFilterReducerState;
+};
+export interface TableFilterReducerState {
+  column: string;
+  database: FilterCategories;
+  schema: string;
+  table: string;
+  tag: string;
```

Comment:
Isn't this tightly coupled with the resource config? It should be possible to use generic types, maybe something like: 

```
interface MultiFilter {
  category: string;
  values: string[];
}
interface SingleFliter {
 category: [];
 value: string;
}

export interface TableFilterReducerState {
  multiFilters: MultiFilter[];
  singleFilters: SingleFilter[];
}
```

Or we could make these objects rather than arrays

```
interface SingleFilters {
  [category: string]: string
}

interface MultiFIlters {
  [category: string]: string[];
}
```

Reply:
I will explore the second option, I agree it should be improved. This version came from going in circles a bit too long on this part and hardcoding things to work.